pkg/eestream/rs_test.go

```diff
@@ -43,7 +43,7 @@ func TestRS(t *testing.T) {
 		readerMap[i] = ioutil.NopCloser(reader)
 	}
 	decoder := DecodeReaders(ctx, readerMap, rs, 32*1024, 0)
-	defer decoder.Close()
+	defer func() { assert.NoError(t, decoder.Close()) }()
```

Comment:
This is neat! Does only one instruction need to be wrapped in an anonymous function?

Reply:
Every returned error must be explicitly checked or ignored.

Reply:
curious could you do `defer assert.NoError(t, decoder.Close())` ?

Reply:
No, because then `decoder.Close()` would be immediately resolved instead of when the `defer` is resolved.